balancer/endpointsharding/endpointsharding_test.go

```diff
@@ -157,3 +162,54 @@ func (s) TestEndpointShardingBasic(t *testing.T) {
 		t.Fatalf("error in expected round robin: %v", err)
 	}
 }
+
+// TestEndpointShardingStuckConnecting verifies that the endpointsharding policy
+// handles child polcies that haven't given a picker update correctly and doesn't
+// panic.
+func (s) TestEndpointShardingStuckConnecting(t *testing.T) {
+	childPolicyName := t.Name()
+	stub.Register(childPolicyName, stub.BalancerFuncs{
+		UpdateClientConnState: func(_ *stub.BalancerData, ccs balancer.ClientConnState) error {
+			t.Logf("Ignoring resolver update to remain in CONNECTING: %v", ccs)
+			return nil
+		},
+	})
+	childLbJSON := json.RawMessage(fmt.Sprintf(`[{%q: {}}]`, childPolicyName))
+	var parseErr error
+	childLBConfig, parseErr = ParseConfig(childLbJSON)
+	if parseErr != nil {
+		t.Fatalf("Failed to parse child LB config: %v", parseErr)
+	}
+	backend1 := stubserver.StartTestService(t, nil)
+	defer backend1.Stop()
+	backend2 := stubserver.StartTestService(t, nil)
+	defer backend2.Stop()
+
+	mr := manual.NewBuilderWithScheme("e2e-test")
+	defer mr.Close()
+
+	json := `{"loadBalancingConfig": [{"fake_petiole":{}}]}`
+	sc := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(json)
+	mr.InitialState(resolver.State{
+		Endpoints: []resolver.Endpoint{
+			{Addresses: []resolver.Address{{Addr: backend1.Address}}},
+			{Addresses: []resolver.Address{{Addr: backend2.Address}}},
+		},
+		ServiceConfig: sc,
+	})
+
+	cc, err := grpc.Dial(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		log.Fatalf("Failed to dial: %v", err)
+	}
+	defer cc.Close()
+	ctx, cancel := context.WithTimeout(context.Background(), defaultShortTestTimeout)
+	defer cancel()
+	client := testgrpc.NewTestServiceClient(cc)
+
+	// Even though the child LB policy hasn't given an picker updates, it is
```

Comment:
> // Even though the child LB policy hasn't given an picker updates, it is assumed that its in connecting state

We should verify that there is no state change i.e. balancer was created with `connecting` state and balancer is still in `connecting` state. May be check `State.ConnectivityState`?